frontend/src/components/header/Header.tsx

```diff
@@ -334,23 +343,38 @@ export default function Header({
     }
   };
 
-  const logo = getLogo();
-
-  const getLogoLink = () => {
-    if (hubUrl) {
-      return `${localePrefix}/hubs/${hubUrl}`;
+  const makeLogoLink = () => {
+    let logoLink = "/";
+    if (isLandingPage || (isHubPage && hubUrl)) {
```

Comment:
Doesn't this achieve the same with less complexity?

```suggestion
    if (hubUrl) {
```